core/src/main/java/org/fao/geonet/api/records/attachments/FilesystemStore.java

```diff
@@ -79,7 +80,15 @@ public FilesystemStore() {
                                                String filter) throws Exception {
         List<MetadataResource> resourceList = new ArrayList<>();
         ApplicationContext _appContext = ApplicationContextHolder.get();
-        String metadataId = getAndCheckMetadataId(metadataUuid);
+        String metadataId;
+
+        if (StringUtils.isNumeric(metadataUuid)) {
```

Comment:
What happens if someone is using numeric UUIDs?
```
<gmd:fileIdentifier><gco:CharacterString>9999987</...
```

Reply:
You're right, I'll check to update the code as done in other places, checking first if the value exists as uuid and if not take it as id.

Reply:
@fxprunayre I have restored `FilesystemStore` to keep working with metadata uuids and move the management of the id/uuids to the attachments API. I think that is better to maintain.

But there're more deeper issues with uuids in url paths and changing them to use ids. For example online resources/thumbnails will use the id of the metadata in the url added to the metadata, not optimal, but can be fine.

But when exporting the metadata to MEF and importing it to other catalogue, using the metadata id instead of uuid probably will cause issues with the links.

Also the link between services/datasets use the formatter xml service, if using id and re-importing the referenced metadata will cause a metadata id change, so the link reference will be no longer valid.